src/js/controller/qoe.ts

```diff
@@ -86,30 +101,32 @@ function trackFirstFrame(model, programController) {
     programController.on(MEDIA_TIME, model._onTime);
 }
 
-const initQoe = function(initialModel, programController) {
-    function onMediaModel(model, mediaModel, oldMediaModel) {
+const initQoe = function(initialModel: Model, programController: ProgramController): void {
+    function onMediaModel(model: QoeModel, mediaModel: MediaModel, oldMediaModel: MediaModel): void {
         // finish previous item
         if (model._qoeItem && oldMediaModel) {
             model._qoeItem.end(oldMediaModel.get('mediaState'));
         }
         // reset item level qoe
-        model._qoeItem = new Timer();
-        model._qoeItem.getFirstFrame = function() {
-            const time = this.between(MEDIA_PLAY_ATTEMPT, MEDIA_FIRST_FRAME);
-            // If time between the tab becoming visible and first frame is valid
-            // and less than the time since play attempt, play was not attempted until the tab became visible
-            const timeActive = this.between(TAB_VISIBLE, MEDIA_FIRST_FRAME);
-            if (timeActive > 0 && timeActive < time) {
-                return timeActive;
+        model._qoeItem = {
+            ...Timer(),
```

Comment:
Interesting fix here - with JS this _should_ be functionally identical but I’m wondering if we should add whatever component has timer tests to the Jira ticket so we get some automation on the change

Reply:
Also should this be new Timer()...?

Reply:
MIGHT be worth subclassing Timer with qoeItem? Might be more trouble than it’s worth but the more I look at this the less easy it is to quickly “get” for me

Reply:
Looking at this more, I think it is worth it to subclass because this could get ugly.
This will now be `new Timer()`, but wasn't here because `Timer` is really just a function that returns a Timer so it was complaining.